include/golioth/golioth_debug.h

```diff
@@ -16,7 +16,7 @@
 #define LOG_TAG_DEFINE(tag) static __UNUSED const char* TAG = #tag
 #endif
 
-typedef void* golioth_client_t;
+struct golioth_client;;
```

Comment:
```suggestion
struct golioth_client;
```

It looks like there is an extra semicolon on his one.